opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -0,0 +1,536 @@
+///
+/// ## Converting from PEM to PKCS#12
+///
+/// If you have PEM format cert and key, you can convert them using OpenSSL:
+///
+/// ### Linux/macOS:
+/// ```bash
+/// openssl pkcs12 -export \
+///   -in cert.pem \
+///   -inkey key.pem \
+///   -out client.p12 \
+///   -name "alias"
+/// ```
+///
+/// ### Windows (PowerShell):
+/// ```powershell
+/// openssl pkcs12 -export -in cert.pem -inkey key.pem -out client.p12 -name "alias"
+/// ```
+#[allow(dead_code)]
+#[derive(Clone)]
+pub(crate) enum AuthMethod {
+    /// Certificate-based authentication
+    ///
+    /// # Arguments
+    /// * `path` - Path to the PKCS#12 (.p12) certificate file
+    /// * `password` - Password to decrypt the PKCS#12 file
+    Certificate { path: String, password: String },
+    /// Azure Managed Identity authentication
+    ///
+    /// Note(TODO): This is not yet implemented.
+    ManagedIdentity,
+}
+
+#[derive(Debug, Error)]
+pub(crate) enum GenevaConfigClientError {
+    #[error("HTTP error: {0}")]
+    Http(#[from] reqwest::Error),
+    #[error("Certificate error: {0}")]
+    Certificate(String),
+    #[error("Missing Auth Info: {0}")]
+    AuthInfoNotFound(String),
+    #[error("Request failed with status {status}: {message}")]
+    RequestFailed { status: u16, message: String },
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+    #[error("TLS error: {0}")]
+    Tls(#[from] native_tls::Error),
+    #[error("JSON error: {0}")]
+    SerdeJson(#[from] serde_json::Error),
+    #[error("Moniker not found: {0}")]
+    MonikerNotFound(String),
+    #[error("Internal error: {0}")]
+    InternalError(String),
+    //#[error("Mismatched TagId. Sent: {sent}, Received: {received}")]
+    //MismatchedTagId { sent: String, received: String },
+}
+
+#[allow(dead_code)]
+pub(crate) type Result<T> = std::result::Result<T, GenevaConfigClientError>;
+
+/// Configuration for the Geneva Config Client.
+///
+/// # Fields
+/// * `endpoint` - The Geneva Config Service endpoint URL
+/// * `environment` - Environment name (e.g., "prod", "dev")
+/// * `account` - Account name in Geneva
+/// * `namespace` - Namespace for the configuration
+/// * `region` - Azure region (e.g., "westus2")
+/// * `config_major_version` - Major version of the configuration schema
+/// * `auth_method` - Authentication method to use (Certificate or ManagedIdentity)
+///
+/// # Example
+/// ```ignore
+/// let config = GenevaConfigClientConfig {
+///     endpoint: "https://example.geneva.com".to_string(),
+///     environment: "prod".to_string(),
+///     account: "myaccount".to_string(),
+///     namespace: "myservice".to_string(),
+///     region: "westus2".to_string(),
+///     config_major_version: 1,
+///     auth_method: AuthMethod::Certificate {
+///         path: "/path/to/cert.p12".to_string(),
+///         password: "password".to_string(),
+///     },
+/// };
+/// ```
+#[allow(dead_code)]
+#[derive(Clone)]
+pub(crate) struct GenevaConfigClientConfig {
+    pub endpoint: String,
```

Comment:
nit: Mark the access modifiers for each of these fields as `pub(crate)` similar to `IngestionGatewayInfo` struct.